### DISCUSSION TITLE
any way to bat (cat or more) the zipped text file like 'less'?

### DISCUSSION THREAD

**Discussion by yaanlpc** [sharkdp/bat#2802] [Ideas]
Hope there would be no additional arguments to 'less' zipped text file

**Comment by eth-p** [comment]
If I understand correctly, you want to know if it's possible to print the text contents of something like `file.txt.gz`? That would be like how `lesspipe` preprocesses files/directories that are given to `less`.

The way `less` does it is by sending the file name to a command specified by the `LESSOPEN` environment variable. A couple years back, we had experimental support for that (#1597) which didn't make it into the `main` branch. More recently, #2444 added proper support for it. It's not provided in default builds yet, however.

In the meantime, it's still possible to use `bat` *as* `LESSOPEN`:

<img width="791" alt="image" src="https://github.com/sharkdp/bat/assets/32112321/cc126a00-8b54-4851-9887-13848562f171">
